### PR TITLE
Rename all 'initialize' methods

### PR DESCRIPTION
**Pull request by ajaybhat**
This is a fix for #2766

### CODE REVIEW

**Comment by ajaybhat** [comment]
@karlnapf Does this look okay to merge?

File: src/shogun/neuralnets/RBM.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ void CRBM::add_visible_group(int32_t num_units, ERBMVisibleUnitType unit_type)
 			m_visible_state_offsets->element(n-1)+m_visible_group_sizes->element(n-1));
 }
 
-void CRBM::initialize(float64_t sigma)
```
Comment (lisitsyn) [inline src/shogun/neuralnets/RBM.cpp]: Lets name it ```initialize_neural_net``` as well. It does initialize weights now but if we change something we'd have to rename it.

Comment (ajaybhat) [inline src/shogun/neuralnets/RBM.cpp]: @lisitsyn  i've fixed it.

Comment (ajaybhat) [inline src/shogun/neuralnets/RBM.cpp]: I've also taken care of some compile errors that kept appearing.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1090,7 +1090,7 @@ IF(DOXYGEN_FOUND)
 		add_subdirectory(${CMAKE_SOURCE_DIR}/doc)
 	ENDIF()
 ENDIF()
-
+add_definitions(-std=c++11)
```
Comment (lisitsyn) [inline CMakeLists.txt]: Why?

**Comment by lisitsyn** [comment]
Sorry, one more issue

**Comment by iglesias** [comment]
@ajaybhat, do you want to get back to it?

File: src/shogun/neuralnets/NeuralLayer.h [hunk 3]
```diff
@@ -106,8 +106,8 @@ class CNeuralLayer : public CSGObject
 	 * @param input_indices  Indices of the layers that are connected to this
 	 * layer as input
 	 */
-	virtual void initialize(CDynamicObjectArray* layers,
-			SGVector<int32_t> input_indices);
+	virtual void initialize_neural_net_layer(CDynamicObjectArray *layers,
+											 SGVector<int32_t> input_indices);
```
Comment (iglesias) [inline src/shogun/neuralnets/NeuralLayer.h]: Something funny with the indentation here.

File: src/shogun/neuralnets/NeuralLinearLayer.h [hunk 4]
```diff
@@ -83,8 +83,8 @@ class CNeuralLinearLayer : public CNeuralLayer
 	 * @param input_indices  Indices of the layers that are connected to this
 	 * layer as input
 	 */
-	virtual void initialize(CDynamicObjectArray* layers,
-			SGVector<int32_t> input_indices);
+	virtual void initialize_neural_net_layer(CDynamicObjectArray *layers,
+											 SGVector<int32_t> input_indices);
```
Comment (iglesias) [inline src/shogun/neuralnets/NeuralLinearLayer.h]: Indentation.

**Comment by erip** [comment]
Should Tapkee `initialize` methods be renamed as well? I wouldn't imagine so because there should exist no interplay between the libraries and Ruby. Further, I don't think it's allowed as per the LICENSE file.

**Comment by lisitsyn** [comment]
@erip I don't think license is issue. As long as they are not exported to swig it should be ok with whatever name

**Comment by erip** [comment]
Recommend closing this.

**Comment by besser82** [comment]
@erip done…

**Pull request closed without merging**